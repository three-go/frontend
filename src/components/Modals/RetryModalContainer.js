import React, { useState } from "react";

import { useNavigation } from "@react-navigation/native";

import { navigations, colors } from "../../common/constants";
import SmallButton from "../Buttons/SmallButton";
import ButtonModal from "../Modals/ButtonModal";

const RetryModalContainer = ({ onRetryGame }) => {
  const [modalVisible, setModalVisible] = useState(true);
  const navigation = useNavigation();

  const handleNavigateMain = () => {
    navigation.navigate(navigations.main);
    setModalVisible(false);
  };

  const handleRetry = () => {
    onRetryGame();
    setModalVisible(false);
  };

  return (
    <ButtonModal isVisible={modalVisible} content="계속 도전 하시겠습니까?">
      <SmallButton
        content="도전하기"
        color={colors.tealGreen}
        onPress={handleRetry}
      />
      <SmallButton
        content="메인으로"
        color={colors.red}
        onPress={handleNavigateMain}
      />
    </ButtonModal>
  );
};

export default RetryModalContainer;
