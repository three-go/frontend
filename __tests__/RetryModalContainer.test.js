import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import RetryModalContainer from "../src/components/Modals/RetryModalContainer";

const mockNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

describe("RetryModalContainer", () => {
  it("RetryModalContainer render check", () => {
    const monkRetryGame = jest.fn();

    const component = render(
      <RetryModalContainer onRetryGame={monkRetryGame} />
    );

    expect(component.getByText("도전하기")).toBeTruthy();
    expect(component.getByText("메인으로")).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
    expect(component.toJSON).toBeTruthy();
  });

  it("RetryModalContainer '도전하기' button click", () => {
    const monkRetryGame = jest.fn();

    const component = render(
      <RetryModalContainer onRetryGame={monkRetryGame} />
    );

    const retryButton = component.getByText("도전하기");
    fireEvent.press(retryButton);

    expect(monkRetryGame.mock.calls.length).toBe(1);
    expect(() => component.getByText("도전하기")).toThrow(
      "Unable to find an element with text: 도전하기"
    );
  });

  it("RetryModalContainer '메인으로' button click", () => {
    const monkRetryGame = jest.fn();

    const component = render(
      <RetryModalContainer onRetryGame={monkRetryGame} />
    );

    const mainNavigateButton = component.getByText("메인으로");
    fireEvent.press(mainNavigateButton);

    expect(mockNavigate.mock.calls.length).toBe(1);
    expect(() => component.getByText("메인으로")).toThrow(
      "Unable to find an element with text: 메인으로"
    );
  });
});
