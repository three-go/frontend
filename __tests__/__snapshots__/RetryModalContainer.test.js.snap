// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetryModalContainer RetryModalContainer render check 1`] = `[Function]`;
